src/v/archival/ntp_archiver_service.cc

```diff
@@ -983,6 +985,79 @@ ss::future<cloud_storage::upload_result> ntp_archiver::upload_tx(
       get_bucket_name(), manifest, fib, _tx_tags);
 }
 
+ss::future<std::optional<cloud_storage::offset_index>>
+ntp_archiver::make_segment_index(
```

Comment:
Yes, the index always needs to match the byte range we upload. It will eventually be used to scan through the data uploaded to cloud store, which will become the downloaded index file in the read path, so generating an index matching the uploaded data should suffice.